modules/webm/video_stream_webm.cpp

```diff
@@ -307,40 +305,19 @@ void VideoStreamPlaybackWebm::update(float p_delta) {
 
 			if (should_process(*video_frame)) {
 
-				if ((err = video->getImage(image)) != VPXDecoder::NO_FRAME) {
-
-					if (err == VPXDecoder::NO_ERROR && image.w == webm->getWidth() && image.h == webm->getHeight()) {
-
-						PoolVector<uint8_t>::Write w = frame_data.write();
-						bool converted = false;
-
-						if (image.chromaShiftW == 1 && image.chromaShiftH == 1) {
-
-							yuv420_2_rgb8888(w.ptr(), image.planes[0], image.planes[2], image.planes[1], image.w, image.h, image.linesize[0], image.linesize[1], image.w << 2, 0);
-							// 								libyuv::I420ToARGB(image.planes[0], image.linesize[0], image.planes[2], image.linesize[2], image.planes[1], image.linesize[1], w.ptr(), image.w << 2, image.w, image.h);
-							converted = true;
-						} else if (image.chromaShiftW == 1 && image.chromaShiftH == 0) {
-
-							yuv422_2_rgb8888(w.ptr(), image.planes[0], image.planes[2], image.planes[1], image.w, image.h, image.linesize[0], image.linesize[1], image.w << 2, 0);
-							// 								libyuv::I422ToARGB(image.planes[0], image.linesize[0], image.planes[2], image.linesize[2], image.planes[1], image.linesize[1], w.ptr(), image.w << 2, image.w, image.h);
-							converted = true;
-						} else if (image.chromaShiftW == 0 && image.chromaShiftH == 0) {
-
-							yuv444_2_rgb8888(w.ptr(), image.planes[0], image.planes[2], image.planes[1], image.w, image.h, image.linesize[0], image.linesize[1], image.w << 2, 0);
-							// 								libyuv::I444ToARGB(image.planes[0], image.linesize[0], image.planes[2], image.linesize[2], image.planes[1], image.linesize[1], w.ptr(), image.w << 2, image.w, image.h);
-							converted = true;
-						} else if (image.chromaShiftW == 2 && image.chromaShiftH == 0) {
-
-							// 								libyuv::I411ToARGB(image.planes[0], image.linesize[0], image.planes[2], image.linesize[2], image.planes[1], image.linesize[1], w.ptr(), image.w << 2, image.w, image.h);
-							// 								converted = true;
-						}
+				{
+					// Scope isolation just in case Write is picky about when it occurs.
+					PoolVector<uint8_t>::Write w = frame_data.write();
+					image.w = webm->getWidth();
+					image.h = webm->getHeight();
+					image.data = w.ptr();
+					err = video->getImage(image);
+				}
```

Comment:
Out of curiousity, what's with the extra set of `{ }` brackets in this section? The `if` statement just above this already has its own `{` opening bracket.

Reply:
Wait nevermind, I misread the comment included at the beginning of this section. Woops!